persistence/postgres-persistence/src/main/java/com/netflix/conductor/postgres/dao/PostgresLockDAO.java

```diff
@@ -0,0 +1,130 @@
+package com.netflix.conductor.postgres.dao;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.netflix.conductor.core.sync.Lock;
+import com.netflix.conductor.model.TaskModel;
+import com.netflix.conductor.postgres.util.Query;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.retry.support.RetryTemplate;
+
+import javax.sql.DataSource;
+import java.util.concurrent.*;
+
+public class PostgresLockDAO extends PostgresBaseDAO implements Lock {
+    private static final Logger LOGGER = LoggerFactory.getLogger(PostgresLockDAO.class);
+    private static final ConcurrentHashMap<String, ScheduledFuture<?>> SCHEDULEDFUTURES =
+            new ConcurrentHashMap<>();
+    private static final ThreadGroup THREAD_GROUP = new ThreadGroup("PostgresLock-scheduler");
+    private static final ThreadFactory THREAD_FACTORY =
+            runnable -> new Thread(THREAD_GROUP, runnable);
+    private static final ScheduledExecutorService SCHEDULER =
+            Executors.newScheduledThreadPool(1, THREAD_FACTORY);
+    protected PostgresLockDAO(RetryTemplate retryTemplate, ObjectMapper objectMapper, DataSource dataSource) {
+        super(retryTemplate, objectMapper, dataSource);
+    }
+
+    @Override
+    public void acquireLock(String lockId) {
+        LOGGER.trace("Acquiring lock {}", lockId);
+        // create code for acquiring advisory lock in postgres
+        final String ADVISORY_LOCK = "SELECT pg_advisory_lock (?)";
+
+        // execute the query
+        queryWithTransaction(ADVISORY_LOCK, (q) -> {
+            q.addParameter(lockId);
```

Comment:
the pg_advisory_lock expects int as input doesn't it ? but we are passing string parameter